lxc/file.go

```diff
@@ -299,6 +299,14 @@ func (c *cmdFilePull) run(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	// Create a slice of funcs to run in a deferred func. Append to this list instead of deferring in the loop below.
```

Comment:
We just need to be careful when replacing defer in a loop with this sort of construct because unlike defer (or our revert package) this is calling the cleanup funcs in the order they were appended not in reverse order.

Is there a reason you didn't go for using the revert package here without a call to revert.Success?

Reply:
I did that initially but felt the naming might be confusing as we are not reverting anything - we want the reverter to fail. Happy to change it back.